parsing/lexer.mll

```diff
@@ -829,12 +860,13 @@ rule token = parse
    indicated by the [already_consumed] argument. The caller is responsible
    for checking that the '#' appears in column 0.
 
-   The [directive] lexer always attempts to read the line number from the
-   lexbuf. It expects to receive a line number from exactly one source (either
-   the lexbuf or the [already_consumed] argument, but not both) and will fail if
-   this isn't the case.
+   The [directive] lexer always attempts to read the line number (or something else,
+   like "syntax" or other directive name) from the lexbuf.
```

Comment:
I think this part of the comment is no longer useful. The point of the original was to explain an invariant around the [already_consumed] parameter which is now explained just below.